### PR TITLE
Improving file stat code

### PR DESCRIPTION
**Pull request by suryagupta4**
# Description
Improving file stat code to log proper error in case of any error while checking stat of the file.

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, vetting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [x] Ran make unit-test
image

### CODE REVIEW

File: service/ephemeral.go [hunk 1]
```diff
@@ -61,13 +61,18 @@ func (s *service) ephemeralNodePublish(
...
+	if err != nil {
+		if os.IsNotExist(err) {
+			Log.Debug("path does not exist, will attempt to create it")
+			err = os.MkdirAll(ephemeralStagingMountPath, 0750)
+			if err != nil {
+				Log.Errorf("ephemeralNodePublish: %s", err.Error())
+				return nil, status.Error(codes.Internal, "Unable to create directory for mounting ephemeral volumes")
+			}
 		}
+		Log.WithField("file", ephemeralStagingMountPath).WithError(
```
Comment (adarsh-dell) [inline service/ephemeral.go:74]: Line 74: Would it be feasible to log the exact same error (that we got at Line no 64)?

File: service/mount.go [hunk 2]
```diff
@@ -550,22 +550,26 @@ func contains(list []string, item string) bool {
...
+				Log.WithField("file", path).WithError(
+					err).Error("Unable to close file")
+			}
+			Log.WithField("path", path).Debug("created file")
+			return true, nil
 		}
-		Log.WithField("path", path).Debug("created file")
-		return true, nil
+		Log.WithField("file", path).WithError(
+			err).Error("Unable to check stat of file")
```
Comment (adarsh-dell) [inline service/mount.go:572]: Same comment as above.

File: service/ephemeral.go (hunk 1 above)
Comment (suryagupta4) [reply service/ephemeral.go]: Updated

File: service/mount.go (hunk 2 above)
Comment (suryagupta4) [reply service/mount.go]: Updated

**Review by adarsh-dell — APPROVED** [review]

**Review by KshitijaKakde — APPROVED** [review]

**Review by adarsh-dell — APPROVED** [review]

**Pull request merged**